scancodeio/wsgi.py

```diff
@@ -33,6 +33,27 @@
 
 from django.core.wsgi import get_wsgi_application
 
+from celery.exceptions import OperationalError
+
+from scancodeio.celery import app
+from scanpipe.models import Run
+
 os.environ.setdefault("DJANGO_SETTINGS_MODULE", "scancodeio.settings")
 
+
+def get_celery_worker_status():
+    try:
+        i = app.control.inspect()
+        active = i.active()
+        scheduled = i.scheduled()
+        if active[list(active.keys())[0]] or scheduled[list(scheduled.keys())[0]]:
```

Comment:
Could there be a way to make this easier to read and understand? may be some intermediate variable? and is there a risk to use the first element? can this fail if the list are empty? Can they be empty? Is  there a difference between `list(active.keys())` and `list(active)` assuming `active` is some kind of mapping?
What are you trying to achieve here? get the first value of either one of two mappings?

Reply:
I will change variable names and try to make it more readable. When testing on my PC, `i.active()` returned a dictionary with one key of the format `celery@<-pc name->`, so I had to use the first key by using `list(active.keys())[0]`.  If no worker is avaliable, it simply returns a dictionary object like `{'celery@avishrant-Latitude-3500': []}`.

`can this fail if the list are empty?` - @pombredanne both active() and scheduled() will always work provided that the broker is reachable.

`What are you trying to achieve here?` - This code snippet simply checks if any job is running or scheduled. If it finds such a job, the Database won't be updated. Alternatively, if no job is running and any 'Run' database entry has `running`  task status, it simply updates that field, thereby solving the problem.